lib/ruby_lsp/setup_bundler.rb

```diff
@@ -282,12 +282,21 @@ def run_bundle_install_directly(env, force_install: false)
 
       return update(env) if @needs_update_path.exist?
 
-      # The ENV can only be merged after checking if an update is required because we depend on the original value of
-      # ENV["BUNDLE_GEMFILE"], which gets overridden after the merge
       FileUtils.touch(@needs_update_path) if needs_update
 
       $stderr.puts("Ruby LSP> Checking if the composed bundle is satisfied...")
-      missing_gems = bundle_check
+
+      begin
+        missing_gems = bundle_check
+      rescue Errno::EPIPE, Bundler::HTTPError
+        # These are errors cases where we cannot recover
+        raise
```

Comment:
We were not re-raising here and instead trying to bundle install again if something failed during bundle check. I refactored the code to scope the rescue to only bundle check and made sure we're re-raising in the cases where we cannot try to recover.